src/bufr/BufrParser/BufrParser.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "BufrParser.h"
+
+#include <iostream>
+#include <map>
+
+#include "bufr.interface.h"
+
+#include "BufrParser/BufrCollectors/BufrCollectors.h"
+#include "BufrMnemonicSet.h"
+#include "DataContainer.h"
+
+#include "Exports/MnemonicExport.h"
+#include "Exports/DatetimeExport.h"
+#include "Exports/Export.h"
+
+
+namespace Ingester
+{
+    static const unsigned int SUBSET_STR_LEN = 25;
+
+    BufrParser::BufrParser(BufrDescription &description) :
+        description_(description),
+        fileUnit_(0)
+    {
+        reset();
+    }
+
+    BufrParser::~BufrParser()
+    {
+        closeBufrFile();
+    }
+
+    std::shared_ptr <DataContainer> BufrParser::parse(const size_t maxMsgsToParse)
+    {
+        assert(fileUnit_ > 0);
```

Comment:
```assert``` is a macro in C that is stripped away during Release builds. [Eckit provides](https://github.com/JCSDA/eckit/blob/release-stable/src/eckit/exception/Exceptions.h) an ```Assert``` function and an ```ASSERT``` macro to call it.

Reply:
I used assert here because a invalid fileUnit would be indicative of a programming error by a developer which could be caught very quickly while debugging without needing the release version to perform this check. Actually it should probably be fileUnit_ > 10... I can use the ECKit Assert instead.

Reply:
Used BadValue exception instead.